forem/forem
# Remove Elasticsearch ✂️ ✂️ ✂️ ✂️ ✂️ ✂️ ✂️ ✂️ ✂️ ✂️ ✂️ ✂️ ✂️ ✂️

## What type of PR is this? (check all applicable)

## Description
This just might be my favorite PR in this repo..._ever_! This PR removes Elasticsearch.

@rhymes and I have coordinated with SRE to deploy this. We're going to deploy to Ben Halpern Community first to see how it goes. This will result in 89723894723894723 Sidekiq errors which we'll delete manually.

In terms of reviewing this PR...here's a list of keywords we looked for to try and catch where it was being used:
- `search`
- `search:`
- `elasticsearch`
- `index`
- `index_to`
- `Search::`

_Please note - with regards to the serializers...we can probably get rid of a few more. But we have to rename the existing ones anyway so I think it will be better to leave them in and clean them up once we do the renaming._

## Related Tickets & Documents
closes https://github.com/forem/forem/issues/13590
closes https://github.com/forem/rfcs/pull/153

## QA Instructions, Screenshots, Recordings
- Fire up the app locally, play around. See if anything appears broken.
- Search the codebase for missed references to Elasticsearch.

## Added tests?

## [Forem core team only] How will this change be communicated?

## [optional] Are there any post deployment tasks we need to perform?
**DO NOT MERGE**
We will coordinate with SRE to deploy this to Ben Halpern Community, monitor everything, and then work on merging it in with `main`. We'll also need to:
- remove Elasticsearch monitors
- manually clear Sidekiq errors.

![lord_of_the_rings_its_gone_gif](https://media.giphy.com/media/11GWLm7bE2fibC/giphy.gif)
